[
  {
    "url": "https://api.github.com/repos/quarnster/SublimeClang/issues/comments/10013765",
    "html_url": "https://github.com/quarnster/SublimeClang/issues/159#issuecomment-10013765",
    "issue_url": "https://api.github.com/repos/quarnster/SublimeClang/issues/159",
    "id": 10013765,
    "user": {
      "login": "quarnster",
      "id": 1230389,
      "avatar_url": "https://secure.gravatar.com/avatar/10a89302c6736c988ce06d8a41c923c0?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png",
      "gravatar_id": "10a89302c6736c988ce06d8a41c923c0",
      "url": "https://api.github.com/users/quarnster",
      "html_url": "https://github.com/quarnster",
      "followers_url": "https://api.github.com/users/quarnster/followers",
      "following_url": "https://api.github.com/users/quarnster/following",
      "gists_url": "https://api.github.com/users/quarnster/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/quarnster/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/quarnster/subscriptions",
      "organizations_url": "https://api.github.com/users/quarnster/orgs",
      "repos_url": "https://api.github.com/users/quarnster/repos",
      "events_url": "https://api.github.com/users/quarnster/events{/privacy}",
      "received_events_url": "https://api.github.com/users/quarnster/received_events",
      "type": "User"
    },
    "created_at": "2012-11-02T13:11:35Z",
    "updated_at": "2012-11-02T13:11:35Z",
    "body": "Please prove that the error is with SublimeClang and not libclang, until then I'm not going to look at this issue."
  },
  {
    "url": "https://api.github.com/repos/quarnster/SublimeClang/issues/comments/10047809",
    "html_url": "https://github.com/quarnster/SublimeClang/issues/159#issuecomment-10047809",
    "issue_url": "https://api.github.com/repos/quarnster/SublimeClang/issues/159",
    "id": 10047809,
    "user": {
      "login": "xiemeilong",
      "id": 2681638,
      "avatar_url": "https://secure.gravatar.com/avatar/82620b7b781a2b4246fc4bc1c839018d?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png",
      "gravatar_id": "82620b7b781a2b4246fc4bc1c839018d",
      "url": "https://api.github.com/users/xiemeilong",
      "html_url": "https://github.com/xiemeilong",
      "followers_url": "https://api.github.com/users/xiemeilong/followers",
      "following_url": "https://api.github.com/users/xiemeilong/following",
      "gists_url": "https://api.github.com/users/xiemeilong/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/xiemeilong/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/xiemeilong/subscriptions",
      "organizations_url": "https://api.github.com/users/xiemeilong/orgs",
      "repos_url": "https://api.github.com/users/xiemeilong/repos",
      "events_url": "https://api.github.com/users/xiemeilong/events{/privacy}",
      "received_events_url": "https://api.github.com/users/xiemeilong/received_events",
      "type": "User"
    },
    "created_at": "2012-11-04T06:03:17Z",
    "updated_at": "2012-11-04T06:13:03Z",
    "body": "@quarnster\r\n\r\nI complie this code using clang with no error and warn, does this can prove that the error is with SublimeClang and not libclang?\r\nI don't know much about libclang. \r\nBut I think even issue is cased by libclang, it shouldn't crash sublime text, SublimeClang should catch it and print error message.\r\n\r\nThanks.\r\n\r\nthe whole code\r\n\r\n\r\n```c++\r\n#include <cppa/cppa.hpp>\r\n#include <Poco/DirectoryIterator.h>\r\nusing namespace std;\r\nusing namespace cppa;\r\nusing namespace Poco;\r\n\r\nclass FileTraceActor : public sb_actor<FileTraceActor> {\r\n\r\n    // grant access to the private init_state member\r\n    friend class sb_actor<FileTraceActor>;\r\n\r\n\r\n    behavior empty = (\r\n        on(atom(\"\"),arg_match)>> [](string parent,string name) {    //crash is case by this line!\r\n            \r\n        }\r\n    );\r\n\r\n  \r\n    behavior& init_state = empty;\r\n\r\n};\r\n\r\n\r\nint main(int argc, char const *argv[])\r\n{\r\n\t/* code */\r\n\treturn 0;\r\n}\r\n\r\n"
  },
  {
    "url": "https://api.github.com/repos/quarnster/SublimeClang/issues/comments/10048128",
    "html_url": "https://github.com/quarnster/SublimeClang/issues/159#issuecomment-10048128",
    "issue_url": "https://api.github.com/repos/quarnster/SublimeClang/issues/159",
    "id": 10048128,
    "user": {
      "login": "quarnster",
      "id": 1230389,
      "avatar_url": "https://secure.gravatar.com/avatar/10a89302c6736c988ce06d8a41c923c0?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png",
      "gravatar_id": "10a89302c6736c988ce06d8a41c923c0",
      "url": "https://api.github.com/users/quarnster",
      "html_url": "https://github.com/quarnster",
      "followers_url": "https://api.github.com/users/quarnster/followers",
      "following_url": "https://api.github.com/users/quarnster/following",
      "gists_url": "https://api.github.com/users/quarnster/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/quarnster/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/quarnster/subscriptions",
      "organizations_url": "https://api.github.com/users/quarnster/orgs",
      "repos_url": "https://api.github.com/users/quarnster/repos",
      "events_url": "https://api.github.com/users/quarnster/events{/privacy}",
      "received_events_url": "https://api.github.com/users/quarnster/received_events",
      "type": "User"
    },
    "created_at": "2012-11-04T07:18:40Z",
    "updated_at": "2012-11-04T07:18:40Z",
    "body": ">I complie this code using clang with no error and warn, does this can prove that the error is with SublimeClang and not libclang?\r\n\r\nNo, clang is not libclang. The stacktrace you gave me only contains entries in libclang with nothing else to point the finger at SublimeClang specifically.\r\n\r\n>But I think even issue is cased by libclang, it shouldn't crash sublime text, SublimeClang should catch it and print error message.\r\n\r\nOr, you know, Sublime Text 2 should catch it and not crash.\r\n\r\n>the whole code\r\n\r\nSorry, but I'm not doing your homework for you. Please prove that it's an issue with SublimeClang and not libclang."
  },
  {
    "url": "https://api.github.com/repos/quarnster/SublimeClang/issues/comments/12616336",
    "html_url": "https://github.com/quarnster/SublimeClang/issues/159#issuecomment-12616336",
    "issue_url": "https://api.github.com/repos/quarnster/SublimeClang/issues/159",
    "id": 12616336,
    "user": {
      "login": "forivall",
      "id": 760204,
      "avatar_url": "https://secure.gravatar.com/avatar/9c9f177fae7d1e9b85fb62d55afdc3ea?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png",
      "gravatar_id": "9c9f177fae7d1e9b85fb62d55afdc3ea",
      "url": "https://api.github.com/users/forivall",
      "html_url": "https://github.com/forivall",
      "followers_url": "https://api.github.com/users/forivall/followers",
      "following_url": "https://api.github.com/users/forivall/following",
      "gists_url": "https://api.github.com/users/forivall/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/forivall/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/forivall/subscriptions",
      "organizations_url": "https://api.github.com/users/forivall/orgs",
      "repos_url": "https://api.github.com/users/forivall/repos",
      "events_url": "https://api.github.com/users/forivall/events{/privacy}",
      "received_events_url": "https://api.github.com/users/forivall/received_events",
      "type": "User"
    },
    "created_at": "2013-01-23T19:07:15Z",
    "updated_at": "2013-01-23T19:07:50Z",
    "body": "I was just getting a similar problem, where libclang segfaulted, causing sublimetext to crash. One solution/workaround I can think of would be to log every call to libclang, and then check the log on startup to see if there was a crash, and display a message. (and there would be a setting to enable/disable this).\r\n\r\nIn my case, I used `clang --analyze`, corrected all of the reported warnings, and then libclang stopped segfaulting. But it may happen again, and my proposed solution would help to at least alert the user to these issues."
  },
  {
    "url": "https://api.github.com/repos/quarnster/SublimeClang/issues/comments/12631673",
    "html_url": "https://github.com/quarnster/SublimeClang/issues/159#issuecomment-12631673",
    "issue_url": "https://api.github.com/repos/quarnster/SublimeClang/issues/159",
    "id": 12631673,
    "user": {
      "login": "xiemeilong",
      "id": 2681638,
      "avatar_url": "https://secure.gravatar.com/avatar/82620b7b781a2b4246fc4bc1c839018d?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png",
      "gravatar_id": "82620b7b781a2b4246fc4bc1c839018d",
      "url": "https://api.github.com/users/xiemeilong",
      "html_url": "https://github.com/xiemeilong",
      "followers_url": "https://api.github.com/users/xiemeilong/followers",
      "following_url": "https://api.github.com/users/xiemeilong/following",
      "gists_url": "https://api.github.com/users/xiemeilong/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/xiemeilong/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/xiemeilong/subscriptions",
      "organizations_url": "https://api.github.com/users/xiemeilong/orgs",
      "repos_url": "https://api.github.com/users/xiemeilong/repos",
      "events_url": "https://api.github.com/users/xiemeilong/events{/privacy}",
      "received_events_url": "https://api.github.com/users/xiemeilong/received_events",
      "type": "User"
    },
    "created_at": "2013-01-24T00:53:24Z",
    "updated_at": "2013-01-24T00:53:45Z",
    "body": "@forivall \r\nthis [bug](http://llvm.org/bugs/show_bug.cgi?id=14252) has been fixed in clang. "
  },
  {
    "url": "https://api.github.com/repos/quarnster/SublimeClang/issues/comments/13057129",
    "html_url": "https://github.com/quarnster/SublimeClang/issues/159#issuecomment-13057129",
    "issue_url": "https://api.github.com/repos/quarnster/SublimeClang/issues/159",
    "id": 13057129,
    "user": {
      "login": "forivall",
      "id": 760204,
      "avatar_url": "https://secure.gravatar.com/avatar/9c9f177fae7d1e9b85fb62d55afdc3ea?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png",
      "gravatar_id": "9c9f177fae7d1e9b85fb62d55afdc3ea",
      "url": "https://api.github.com/users/forivall",
      "html_url": "https://github.com/forivall",
      "followers_url": "https://api.github.com/users/forivall/followers",
      "following_url": "https://api.github.com/users/forivall/following",
      "gists_url": "https://api.github.com/users/forivall/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/forivall/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/forivall/subscriptions",
      "organizations_url": "https://api.github.com/users/forivall/orgs",
      "repos_url": "https://api.github.com/users/forivall/repos",
      "events_url": "https://api.github.com/users/forivall/events{/privacy}",
      "received_events_url": "https://api.github.com/users/forivall/received_events",
      "type": "User"
    },
    "created_at": "2013-02-03T22:45:02Z",
    "updated_at": "2013-02-03T22:45:02Z",
    "body": "@xiemeilong\r\nThere's still other bugs that cause segfaults\r\n@quarnster - you can ignore my suggestion for now - sublimetext3 is adding out of process extensions, so clang won't kill the main sublime process when it segfaults. I'd just wait for that. Yay!"
  }
]
